CONTRIBUTING.md

```diff
@@ -27,36 +27,55 @@ npm install
+
+```bash
+bin/dev install {addon-name} --project {project-name}
+```
+
 ### Project structure
 
-The project has the following main folders:
-
-- `bin` - contains the executable file for the CLI
-  - `dev` - the executable file for the CLI in development mode
-  - `run` - the executable file for the CLI in production mode
-
-- `src` - contains the source code of the CLI
-  - `index.ts` - the entry point of the CLI
-  - `commands` - contains the commands of the CLI
-  - `helpers` - contains the helper functions of the CLI
-    - `childProcess` - contains the helper functions for child process
-    - `file` - contains the helper functions for file interaction
-    - `terraform` - contains the helper functions for Terraform
-  - `templates` - contains the code to generate the templates
-    - `addons` - contains the code to generate the common addons e.g. version control, CI/CD, etc.
-    - `aws` - contains the code to generate the AWS modules e.g. VPC, RDS, etc.
-    - `core` - contains the code to generate the main Terraform files e.g. `main.tf`, `variables.tf`, etc.
-  - `skeleton` - contains the skeleton files for the project
-    - `addons` - contains the skeleton files for common addons e.g. version control, CI/CD, etc.
-    - `aws` - contains the skeleton files for AWS modules
-    - `core` - contains the skeleton files for common files e.g. `main.tf`, `variables.tf`, `.gitignore`, etc.
-  - `test` - contains the test helpers and configurations
+The project has the following main files and folders:
+
+```bash
+├── bin # the executable file for the CLI
+│   ├── dev # the executable file for the CLI in development mode
+│   └── run # the executable file for the CLI in production mode
+├── skeleton # the skeleton files for the project
+│   ├── addons # the skeleton files for common addons e.g. version control, CI/CD, etc.
+│   │   ├── aws # the skeleton files for AWS modules
+│   │   └── versionControl # the skeleton files for version control
+│   └── core # the skeleton folders
+│       ├── base # the skeleton files for the base folder
+│       └── shared # the skeleton files for the shared folder
+├── src # the source code of the CLI
+│   ├── commands # the commands of the CLI
+│   │   ├── generate
+│   │   └── installAddon
+│   ├── helpers # the helper functions of the CLI
+│   │   ├── childProcess.ts
+│   │   ├── file.ts
+│   │   └── terraform.ts
+│   ├── hooks # the hooks of the CLI
+│   │   └── postProcess.ts
+│   ├── index.ts # the entry point of the CLI
+│   └── templates # the code to generate the templates
+│       ├── addons # the code to generate the common addons e.g. version control, CI/CD, AWS, etc.
+│       └── core # the code to generate the main Terraform files e.g. `main.tf`, `variables.tf`, etc.
+└── test # the test helpers and configurations
+    └── matchers # the custom matchers for Jest
+        ├── file.ts
+        ├── index.d.ts
+        └── index.ts
+```
```

Comment:
Can we explain in a short line, what is the difference between the `skeleton` and the `templates` folders?
I might be wrong, but from my understanding:
- `skeleton` are the addon's files/folders ready to be copy/pasted directly into the generated project (if the related addon has been selected)
- `templates` contains the logic of what files must be copied from the `templates` folder, as well as what other instructions must be inserted into the core project files.

(let me know if I misunderstood 🙇 )

Reply:
Yes, you understand it correctly:
- The skeleton folder contains addon-related files and folders that can be directly copied or merged into the generated project.
- The templates folder contains the logic and instructions for generating and modifying core project files during the project generation process.
Let me add it into this as well 🙏

Reply:
ok got it. 

Very optional suggestion: in case the "template" naming is not a convention from a lib, should we rename it with "generator", and "skeleton" with "template"?

As from my initial thought, the "templates" folder must contain "static template files" that will be reused when generating the project. And the logic that will generate the project is... a generator? 🤔

Reply:
(we can do that in a future PR, as it's not directly related with this PR 😇 So that your can merge that one first ^^)